### PR TITLE
Fix(UI) radar star color

### PR DESCRIPTION
**Pull request by Zitchas**
**Feature:** This PR implements the feature request detailed and discussed in issue #{{insert number}}

This PR is for testing and figuring out what I'm doing and asking for help.

Here's what I've got so far:
- interfaces.txt has a section that specifies colors. Currently, the "special" color is the one that dictates the star color. I've changed it here for identification purposes. I have added a "radar star" line with another color. (it isn't actually used)
- radar.cpp has a list of constants that appears to be assigning each type of radar object a number. Or perhaps objects already have a number and what they are is being identified. I have added `Radar::STAR = 9` to the list.
- radar.cpp also has a const Color section that I think is pulling the colors from gamedata, which I think means the interfaces.txt file. I have added a line for "radar star" to the list.
- radar.cpp has a section for drawing the radar that mentions circles without an inner. This is how planets and stars are drawn, so this could be a clue as to where the object is identified. Made a comment to this effect to help find it.
- System.cpp has a section that looks like it is identifying what things are, and has a `object.isStar =` line, but I don't know what it does. I added a comment to it so that it would be easy to find. Another section talked about using "friend system" to identify stuff.

### CODE REVIEW

File: source/Radar.cpp [hunk 1]
```diff
@@ -51,7 +52,9 @@ void Radar::SetCenter(const Point &center)
 }
 
 
-
+// Planets and stars both are circles, so they have "inner" as 0, but I can't
+// tell where it sets that. This may be clue to find where the identity of an object
+// is identified as a star.
 // Add an object. If "inner" is 0 it is a dot; otherwise, it is a ring. The
 // given position should be in world units (not shrunk to radar units).
 void Radar::Add(int type, Point position, double outer, double inner)
```
Comment (RisingLeaf) [inline source/Radar.cpp:60]: inner is set here, whenever the function is called for a planet,
engine.cpp line 2271

File: source/System.cpp [hunk 2]
```diff
@@ -915,7 +915,7 @@ void System::LoadObject(const DataNode &node, Set<Planet> &planets, int parent)
 }
 
 
-
+// This looks relevant to setting if an object is a star, but it has no explanation.
 void System::LoadObjectHelper(const DataNode &node, StellarObject &object, bool removing)
```
Comment (RisingLeaf) [inline source/System.cpp:919]: look in line 928

File: source/Radar.cpp [hunk 3]
```diff
@@ -33,6 +33,7 @@ const int Radar::SPECIAL = 5;
 const int Radar::ANOMALOUS = 6;
 const int Radar::BLINK = 7;
 const int Radar::VIEWPORT = 8;
+const int Radar::STAR = 9;
```
Comment (RisingLeaf) [inline source/Radar.cpp:36]: you also need to add this to radar.h, look in line 39

File: source/Radar.cpp [hunk 4]
```diff
@@ -145,6 +148,7 @@ const Color &Radar::GetColor(int type)
 		*GameData::Colors().Get("radar anomalous"),
 		*GameData::Colors().Get("radar blink"),
 		*GameData::Colors().Get("radar viewport")
```
Comment (RisingLeaf) [inline source/Radar.cpp:150]:
```suggestion
		*GameData::Colors().Get("radar viewport"),
```
dont forget the comma ;)

**Comment by RisingLeaf** [comment]
>     * radar.cpp has a list of constants that appears to be assigning each type of radar object a number. Or perhaps objects already have a number and what they are is being identified. I have added `Radar::STAR = 9` to the list.

As mentioned in my review comment, you can not just define variables inline, if you add new stuff to a class, in this case Radar, always look if you also need to change stuff in the header
Common cases for having to changes something in the header are:

- creating a new class member (variables / functions)
- modifying the parameters a function takes, parameters are the variables between the braces after a function name
 
>     * System.cpp has a section that looks like it is identifying what things are, and has a `object.isStar =` line, but I don't know what it does. I added a comment to it so that it would be easy to find. Another section talked about using "friend system" to identify stuff.

It checks for the location of the sprite, if it is in the star folder, it is a star.
It does that by checking if the first five characters are "star/"

File: source/Radar.cpp (hunk 4 above)
Comment (Zitchas) [reply source/Radar.cpp:150]: Yep. That, at least, I managed to figure out from the debug... Thanks!

File: source/Radar.cpp (hunk 3 above)
Comment (Zitchas) [reply source/Radar.cpp:36]: Alright, got it.

File: source/Radar.cpp (hunk 1 above)
Comment (RisingLeaf) [reply source/Radar.cpp:60]: depending on which editor you use, you can right click on a function name and select find all references to find out where it is used

Comment (Zitchas) [reply source/Radar.cpp:60]: NP++ And yes, it does have a "find all in folder" option that I have been using extensively to look for key terms and things.

**Comment by RisingLeaf** [comment]
Also, this probably just slipped in, but most of the time you want to separate this unrelated data file changes, makes stuff cleaner.

**Comment by Zitchas** [comment]
Are you referring to the governments and free world changes? As best I can tell, those are the recent commit, and somehow my new branch managed to already have them while my master did not. Not sure how to get rid of those.

**Comment by RisingLeaf** [comment]
Screenshot 2022-12-29 at 2 59 41 PM

dont you have this button on your when you are looking at your repo?

File: data/interfaces.txt [hunk 5]
```diff
@@ -101,10 +101,11 @@ color "radar friendly" .4 .6 1. 0.
 color "radar unfriendly" .8 .8 .4 0.
 color "radar hostile" .85 .3 .2 0.
 color "radar inactive" .4 .4 .4 0.
-color "radar special" 1. 1. 1. 0.
+color "radar special" 0.5 1. 0.5 0.
 color "radar anomalous" .7 0. 1. 0.
 color "radar blink" 1. 1. 1. 0.
 color "radar viewport" 0. .3 0. 0.
+color "radar star" 0.7 1. 1.
```
Comment (RisingLeaf) [inline data/interfaces.txt:108]:
```suggestion
color "radar star" 0.7 1. 1. 0.
```
you need an alpha value here

**Comment by Zitchas** [comment]
I do have that sync fork button, but it says that I'm up to date with master.

**Comment by RisingLeaf** [comment]
If you want the new color to be used, you will need to edit StellarObject.cpp l.101

**Comment by Zitchas** [comment]
Alright, in order to ditch the unwanted weird commit changes I swapped the relevant changes to a new branch, located here #130 . Thanks for your help!

**Pull request closed without merging**